### PR TITLE
Add cherry_pick_pull.sh

### LINKED ISSUE #3922: Feat: Automatic cherry pick
**What's it about?**

We discussed that we will speed up the release rhythm, Kubernetes has a standard release process, maybe we can copy their practices.

>  https://github.com/kubernetes/community/blob/master/contributors/devel/sig-release/release.md
>  https://github.com/kubernetes/community/blob/master/contributors/devel/sig-release/cherry-picks.md
>  https://github.com/kubernetes/kubernetes/blob/master/hack/cherry_pick_pull.sh

**What's the reason why we need it?**

https://github.com/kubernetes/kubernetes/issues/10730

/kind feature-request

### PR DESCRIPTION
**Pull request by wansir**
Signed-off-by: hongming <talonwan@yunify.com>

**What type of PR is this?**

Follow  K8s cherry-pick process.

**What this PR does / why we need it**:

This script can help us create a cherry-pick PR automatically.

**Which issue(s) this PR fixes**:

Fixes #3922

### CODE REVIEW

**Comment by yuswift** [comment]
I would like to know how to use this script? Shall we use it like this `hack/cherry_pick_pull.sh upstream/release-3.14 98765
` ?

**Comment by zryfish** [comment]
Yeah, please add documentation and link

**Comment by wansir** [comment]
Developer guide: https://github.com/kubesphere/community/pull/345

File: hack/cherry_pick_pull.sh [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+echo "  git push ${FORK_REMOTE} ${NEWBRANCHUNIQ}:${NEWBRANCH}"
+echo
+read -p "+++ Proceed (anything but 'y' aborts the cherry-pick)? [y/n] " -r
+if ! [[ "${REPLY}" =~ ^[yY]$ ]]; then
+  echo "Aborting." >&2
+  exit 1
+fi
+
+git push "${FORK_REMOTE}" -f "${NEWBRANCHUNIQ}:${NEWBRANCH}"
+make-a-pr
```
Comment (zryfish) [inline hack/cherry_pick_pull.sh:257]: https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline

Comment (zryfish) [reply hack/cherry_pick_pull.sh:257]: BTW, did we test this script on KubeSphere repo?

Comment (wansir) [reply hack/cherry_pick_pull.sh:257]: [This PR](https://github.com/kubesphere/kubesphere/pull/3923) is created by the script automatically.

```
hack/cherry_pick_pull.sh upstream/release-3.1 3796
```

Comment (wansir) [reply hack/cherry_pick_pull.sh]: > https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline

fixed

**Pull request merged**